vyper/parser/global_context.py

```diff
@@ -233,7 +236,10 @@ def get_nonrentrant_counter(self, key):
         if key in self._nonrentrant_keys:
             return self._nonrentrant_keys[key]
         else:
-            counter = len(self._globals) + self._nonrentrant_counter
+            counter = (
+                sum(v.size for v in self._globals.values() if not isinstance(v.typ, MappingType))
```

Comment:
I think this deserves a comment as to what the intent is. The comment directly above this block is still talking about `len(self._globals)`

Reply:
This guy specifically https://github.com/vyperlang/vyper/blob/9380b4710f03346977403fb3219eeddabd34ee66/vyper/parser/global_context.py#L231